docs/pages/fund-apps/estimate-fees.mdx

```diff
@@ -0,0 +1,128 @@
+`}
+/>
+</Zoom>
+
+### Gas fees
+
+Commit messages sent through the **XMTP Appchain** incur **gas fees**. Commit messages require transactions on the XMTP Appchain. Gas fees are paid directly from the payer wallet's XMTP Appchain allowance.
+
+Every commit message an app or agent sends through the XMTP Appchain counts toward its message volume. Commit messages are typically used to perform:
+
+- Group management updated (member additions, permission changes)
+- Identity updates
+- Consent preferences updates
+- Payer-related updates
+
+To learn more, see [Envelope types](/protocol/envelope-types).
+
+Collected gas fees are paid directly to the XMTP Appchain. Fees are used to maintain the Appchain.
+
+## Estimate XMTP fees
+
+Use the [XMTP fee calculator](#TODO) to help you estimate fees to be incurred by your app or agent. This calculator uses network averages, which provides estimates at the correct order of magnitude.
+
+Here are rough estimates of XMTP Broadcast Network and XMTP Appchain fees:
+
+- **Messaging fee**: At launch, a 1 KB text message sent to the XMTP Broadcast Network incurs a fee of ~0.001 USDC (≈ $1 per 1,000 small messages).
+
+- **Gas fee**: A message that triggers a commit to the XMTP Appchain incurs a ≤ 500 gwei fee, which works out to fractions of a cent.
+
+### Pricing and network scale
+
+The network is designed to be profitable at 1B messages/month. As volume grows beyond this point, surplus revenue is recycled to lower the global per-byte price for everyone.
+
+## FAQ about XMTP fees
+
+### What determines the effective price per message?
+
+Messaging fees are charged per byte, with total fees scaling linearly with bytes sent. The effective per-message rate shown in the estimator assumes a 1,024-byte message and includes all components: base fees for message storage, gas fees for metadata written to XMTP’s L3, and a 5% protocol treasury share.
+
+### Is price global or per-sender?
+
+The protocol charges a single global per-byte rate that falls as total network volume rises. All developers benefit from lower per-byte rates as the network grows.
+
+### Are attachments or large media extra?
+
+No. They’re just larger messages. XMTP charges per byte, so sending media costs more only due to size—capped at 1 MB per message. For lower fees and larger files, use remote attachment messages to point to off-network storage like IPFS or S3.
+
+### Is the current fee level sustainable for node operators?
+
+Yes. The network is designed to be profitable at 1B messages/month. As volume grows beyond this point, surplus revenue is recycled to lower the global per-byte price for everyone.
+
+### How are fees estimated for 100M or even 1B users?
+
+At very high volumes, price converges to a global floor that ensures node operators remain sustainable. The current effective floor price is $3 per 100,000 messages. Future node efficiency improvements are expected to push this floor even lower.
```

Comment:
Price will step down over time as the network unlocks and sustains global volume milestones.